src/main.cpp

```diff
@@ -164,11 +175,17 @@ void printlist()
   printf("     %s-e%s                %sDOUBLE%s          E-value threshold                                              %s1%s\n","\033[1m","\033[0m","\033[4m","\033[0m","\033[4m","\033[0m");
   // RAM cannot support 1GB default
   if ( 1073741824/pagesize_gv > maxpages_gv/2)
-    printf("     %s-m%s                %sINT%s             INT Mbytes for loading the reads into memory               %s%lu%s\n","\033[1m","\033[0m","\033[4m","\033[0m","\033[4m",((pagesize_gv*(maxpages_gv/2))/1048576),"\033[0m");
+    printf("     %s-m%s                %sINT%s             INT Mbytes for loading reads in memory with mmap             %s%lu%s\n","\033[1m","\033[0m","\033[4m","\033[0m","\033[4m",((pagesize_gv*(maxpages_gv/2))/1048576),"\033[0m");
   // RAM can support at least 1GB default
   else
-    printf("     %s-m%s                %sINT%s             INT Mbytes for loading the reads into memory                   %s1024%s\n","\033[1m","\033[0m","\033[4m","\033[0m","\033[4m","\033[0m");
+    printf("     %s-m%s                %sINT%s             INT Mbytes for loading reads in memory with mmap               %s[suggested: 1024]%s\n","\033[1m","\033[0m","\033[4m","\033[0m","\033[4m","\033[0m");
   printf("                                        (maximum -m INT is %lu)\n",(((maxpages_gv/2)*pagesize_gv)/1048576));
+  printf("                                        (NOTE: If this option is chosen, reads will be loaded using\n");
+  printf("                                         mmap rather than the default generic stream buffer; if the\n");
+  printf("                                         input reads are in compressed format, this option will be\n");
```

Comment:
It may be possible with Boost C++ libraries but didn't want to add another external dependency. Also, since mmap maps pages into memory, at the moment we have a complex algorithm to parse sequences overlapping two pages. I would like to stay away from that for compressed files :|